### DISCUSSION TITLE
[repro provided] Cannot extend model defaults?

### DISCUSSION THREAD

**Discussion by addlistener** [GrapesJS/grapesjs#5846] [Q&A] [answered]
As disccussed here we should be able to do this
https://github.com/GrapesJS/grapesjs/issues/1374#issuecomment-420387779

My code:
```
    editor.DomComponents.addType(type, {
      // extend: type,
      model: item.model.extend({
        defaults: {
          ...item.model.prototype.defaults,
          highlightable: false,
        },
      }),
    });
```
https://stackblitz.com/edit/grapesjs-react-custom-ui-oqoqcp?file=src%2FpatchNativeComponents.ts

The error
<img width="1316" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/4531670/f733dc42-59c3-4bc6-b8e8-38a9832df6bd">

**Comment by artf** [comment]
Yeah the API here is slightly changed
```js
editor.DomComponents.addType(type, {
        model: {
            defaults: {
                highlightable: false,
            }
        },
 });
```

**Reply by addlistener** [reply]
So that means we cannot extend a component while changing the defaults? 

Not a big issue tho I can always make some hack in `init()`

**Reply by artf** [reply, marked as answer]
> So that means we cannot extend a component while changing the defaults?

It automatically extends parent properties, so you only need to indicate your new/updated properties.
